pkg/reconciler/tenancy/workspace/workspace_reconcile.go

```diff
@@ -97,14 +97,21 @@ func (c *Controller) reconcile(ctx context.Context, ws *tenancyv1alpha1.Workspac
 		return indexers.ByPathAndName[*tenancyv1alpha1.WorkspaceType](tenancyv1alpha1.Resource("workspacetypes"), c.globalWorkspaceTypeIndexer, path, name)
 	}
 
+	// create external client that goes through the front-proxy
+	externalConfig := restclient.CopyConfig(c.externalLogicalClusterAdminConfig)
+	kcpExternalClient, err := kcpclientset.NewForConfig(externalConfig)
```

Comment:
curiosity question: why does `kcpExternalClient` need to be instantiated at every reconciliation run? can't it be a cached member of `*Controller`?

Reply:
Thanks, you're right - I'll move it back to `workspace_controller.go` :+1: